### ISSUE TITLE
Default import name being used to determine if export is used

### ISSUE THREAD

**Issue by renato-bohler** [webpro-nl/knip#646]
It seems like the way we name default imports affects whether Knip reports it as an unused export or not when it sounds like it shouldn't.

Here's a simple repro:

https://stackblitz.com/edit/github-rdchin?file=index.ts,utils.ts

```tsx
// utils.ts
export const utilOne = () => console.log('utilOne');
const utilTwo = () => console.log('utilTwo');
export default utilTwo;

// index.ts
import utilOne from './utils';
//     ^ notice how this is actually `utilTwo` in disguise
utilOne();
```

Knip should report `utilOne` as unused, but it reports no issues:

![image](https://github.com/webpro/knip/assets/25781956/7c067ded-d292-48eb-90c5-7d12f09dd82e)

**Comment by renato-bohler** [comment]
By the way, renaming the default import name makes Knip report the export as unused when functionally nothing has changed:

![image](https://github.com/webpro/knip/assets/25781956/a4c88711-a230-4695-99a6-5355017801f5)

**Comment by webpro** [comment]
Thanks for the report! Iiuc in short: bug exposes (only) if the default import identifier equals existing named export

**Comment by webpro** [comment]
Would be great if you could try this out on your project:

```
npm install -D knip@canary
```

**Comment by renato-bohler** [comment]
Confirmed it works as expected on my project using the canary version. Thank you!

**Comment by webpro** [comment]
Should be fixed in [v5.17.0](https://github.com/webpro-nl/knip/releases/tag/5.17.0)
